src/transports/ipc.rs

```diff
@@ -158,52 +158,52 @@ async fn run_server(unix_stream: UnixStream, messages_rx: mpsc::UnboundedReceive
     let mut socket_reader = reader_stream(socket_reader);
     let mut messages_rx = messages_rx.fuse();
     let mut read_buffer = vec![];
+    let mut closed = false;
 
-    loop {
+    while !closed || pending_response_txs.len() > 0 || subscription_txs.len() > 0 {
```

Comment:
I feel this would be blocking the server anyway, since I think we don't really automatically unsubscribe, no?

I'm not entirely sure what the right behaviour should be though, but I feel that being able to close the client upon drop should disregard all subscriptions that are still open, we should wait for any incomplete requests though. I'm not 100% sure about this, interested about your motivation on this.

```suggestion
    while !closed || pending_response_txs.len() > 0 {
```

Reply:
I agree with you entirely, it'd be best not to count the subscriptions since they're not automatic. That said, thinking about this, me and @FelipeRosa thought it'd be cool to rely on the channel state of subscriptions in order to clean up automatically the subscriptions instead of error-ing it out.

We added the 4593a2f commit that implements this: if a subscription channel is closed, it automatically unsubscribes, which means the check against open subscriptions would be better. In case the user of the client drops the clients but keep the receivers of subscriptions, server remains alive for as long as it holds the receiver.

PS: There's a flaw on this, which is to say that the auto cleaning just happens upon getting a message. We realised this, but thought it was a small trade off for having all the other cases work accordingly.

What do you think?

Reply:
First of all, apologies for the delay! I've been putting this off for way too long. I don't think this is a good solution though, there might be subscriptions that fire only very rarely and this would make closing quite unpredictable.

Why is receiving subscriptions important in case the server is closing? The nature of subscriptions should be ephemeral anyway - i.e. they receive a bunch of notifications, but there are no super strict guarantees about them. 

IMHO if the server is closing up, we should just complete any requests that are still pending and just finish up, despite subscriptions being active, but I'm happy to hear your arguments :)

Reply:
@tomusdrw No arguments here, I agree with you. Both are valid approaches, so I changed the branch to reflect what you suggested. (: